### PR TITLE
convert an http_types::Error into an anyhow::Error

### PR DESCRIPTION
**Pull request by jbr**
Related to https://github.com/http-rs/surf/issues/86

### CODE REVIEW

File: src/error.rs [hunk 1]
```diff
@@ -163,7 +169,6 @@ impl<E: Into<anyhow::Error>> From<E> for Error {
         Self::new(StatusCode::InternalServerError, error)
     }
 }
-
```
Comment (Fishrock123) [inline src/error.rs:166]: this seems accidental

**Review by Fishrock123 — APPROVED** [review]
Seems reasonable.

**Comment by faulesocke** [comment]
The only thing required for interop with anyhow and similar crates is an implementation of `std::error::Error`. No dependency on anyhow or anything else is required.

**Comment by Fishrock123** [comment]
@jbr Any reason you haven't merged this?

**Comment by jbr** [comment]
Nope just keep forgetting to remove the newline you pointed out when I'm at a keyboard. Meeting as-is

**Pull request merged**